test/e2e/kdump_test.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/openshift/machine-config-operator/pkg/daemon/constants"
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/openshift/machine-config-operator/test/helpers"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestKdumpEnablement(t *testing.T) {
+	cs := framework.NewClientSet("")
+
+	// Create infra pool to roll out MC changes
+	unlabelFunc := helpers.LabelRandomNodeFromPool(t, cs, "worker", "node-role.kubernetes.io/infra")
+	helpers.CreateMCP(t, cs, "infra")
+
+	// create old mc to have something to verify we successfully rolled back
+	oldInfraConfig := helpers.CreateMC("old-infra", "infra")
+	_, err := cs.MachineConfigs().Create(context.TODO(), oldInfraConfig, metav1.CreateOptions{})
+	require.Nil(t, err)
+	oldInfraRenderedConfig, err := helpers.WaitForRenderedConfig(t, cs, "infra", oldInfraConfig.Name)
+	err = helpers.WaitForPoolComplete(t, cs, "infra", oldInfraRenderedConfig)
+	require.Nil(t, err)
+
+	// create kargs MC
+	kdumpEnabled := true
+	kargsMC := &mcfgv1.MachineConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   fmt.Sprintf("99-worker-kdump-%s", uuid.NewUUID()),
+			Labels: helpers.MCLabelForRole("infra"),
+		},
+		Spec: mcfgv1.MachineConfigSpec{
+			Config: runtime.RawExtension{
+				Raw: helpers.MarshalOrDie(&ign3types.Config{
+					Ignition: ign3types.Ignition{
+						Version: ign3types.MaxVersion.String(),
+					},
+					Systemd: ign3types.Systemd{
+						Units: []ign3types.Unit{
+							{
+								Name:    "kdump.service",
+								Enabled: &kdumpEnabled,
+							},
+						},
+					},
+				}),
+			},
+			KernelArguments: []string{"crashkernel=300M"},
+		},
+	}
+
+	_, err = cs.MachineConfigs().Create(context.TODO(), kargsMC, metav1.CreateOptions{})
+	require.Nil(t, err)
+	t.Logf("Created %s", kargsMC.Name)
+	renderedConfig, err := helpers.WaitForRenderedConfig(t, cs, "infra", kargsMC.Name)
+	require.Nil(t, err)
+	err = helpers.WaitForPoolComplete(t, cs, "infra", renderedConfig)
+	require.Nil(t, err)
+
+	// Re-fetch the infra node for updated annotations
+	infraNode := helpers.GetSingleNodeByRole(t, cs, "infra")
+	assert.Equal(t, infraNode.Annotations[constants.CurrentMachineConfigAnnotationKey], renderedConfig)
+	assert.Equal(t, infraNode.Annotations[constants.MachineConfigDaemonStateAnnotationKey], constants.MachineConfigDaemonStateDone)
+	kargs := helpers.ExecCmdOnNode(t, cs, infraNode, "cat", "/rootfs/proc/cmdline")
+
+	if !strings.Contains(kargs, "crashkernel=300M") {
+		t.Fatalf("Missing %q in kargs: %q", "crashkernel=300M", kargs)
+	}
+	t.Logf("Node %s has expected craskkernel karg", infraNode.Name)
+
+	helpers.ExecCmdOnNode(t, cs, infraNode, "/bin/sh", "-c", string("chroot /rootfs systemctl reboot"))
```

Comment:
why are we manually doing node reboot instead of using a MachineConfig? Is this how user is supposed to be enabling kdump?